clang/lib/Basic/Targets/RISCV.cpp

```diff
@@ -486,3 +486,15 @@ bool RISCVTargetInfo::validateCpuSupports(StringRef Feature) const {
 bool RISCVTargetInfo::isValidFeatureName(StringRef Name) const {
   return llvm::RISCVISAInfo::isSupportedExtensionFeature(Name);
 }
+
+bool RISCVTargetInfo::validateGlobalRegisterVariable(
+    StringRef RegName, unsigned RegSize, bool &HasSizeMismatch) const {
+  if (RegName == "ra" || RegName == "sp" || RegName == "gp" ||
```

Comment:
Should we accept `zero`?

Reply:
`zero` is not writable/allocatable?

Reply:
Yeah it didn't seem useful. You can still do it with "x0" because I didn't think it was worth the extra code to check for it.